snc.sh

```diff
@@ -164,11 +159,37 @@ DYNAMIC_DATA=$(base64 -w0 node-sizing-enabled.env) envsubst < 99_master-node-siz
 export OPENSHIFT_INSTALL_INVOKER="codeReadyContainers"
 export KUBECONFIG=${INSTALL_DIR}/auth/kubeconfig
 
-OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE ${OPENSHIFT_INSTALL} --dir ${INSTALL_DIR} create ignition-configs ${OPENSHIFT_INSTALL_EXTRA_ARGS}
+OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE ${OPENSHIFT_INSTALL} --dir ${INSTALL_DIR} create single-node-ignition-config ${OPENSHIFT_INSTALL_EXTRA_ARGS}
 # mask the chronyd service on the bootstrap node
-cat <<< $(${JQ} '.systemd.units += [{"mask": true, "name": "chronyd.service"}]' ${INSTALL_DIR}/bootstrap.ign) > ${INSTALL_DIR}/bootstrap.ign
-
-${OPENSHIFT_INSTALL} --dir ${INSTALL_DIR} create cluster ${OPENSHIFT_INSTALL_EXTRA_ARGS} || ${OC} adm must-gather --dest-dir ${INSTALL_DIR}
+cat <<< $(${JQ} '.systemd.units += [{"mask": true, "name": "chronyd.service"}]' ${INSTALL_DIR}/bootstrap-in-place-for-live-iso.ign) > ${INSTALL_DIR}/bootstrap-in-place-for-live-iso.ign
+
+# Download the image
+ISO_URL=$(OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE ${OPENSHIFT_INSTALL} coreos print-stream-json | grep location | grep $ARCH | grep iso | cut -d\" -f4)
+curl -L ${ISO_URL} -o ${INSTALL_DIR}/rhcos-live.iso
+
+podman run --privileged --pull always --rm \
+      -v /dev:/dev -v /run/udev:/run/udev -v $PWD:/data \
+      -w /data quay.io/coreos/coreos-installer:release \
+      iso ignition embed --force \
+      --ignition-file ${INSTALL_DIR}/bootstrap-in-place-for-live-iso.ign \
+      ${INSTALL_DIR}/rhcos-live.iso
+
+sudo cp -Z ${INSTALL_DIR}/rhcos-live.iso /var/lib/libvirt/${SNC_PRODUCT_NAME}/rhcos-live.iso
+sudo virt-install \
+    --name ${SNC_PRODUCT_NAME} \
+    --memory ${SNC_CLUSTER_MEMORY} \
+    --vcpus ${SNC_CLUSTER_CPUS} \
+    --os-variant=rhel9-unknown \
+    --network=network:"${SNC_PRODUCT_NAME}",mac=52:54:00:ee:42:e1 \
+    --nographics \
+    --events on_reboot=restart \
+    --cdrom /var/lib/libvirt/${SNC_PRODUCT_NAME}/rhcos-live.iso \
+    --disk path=/var/lib/libvirt/${SNC_PRODUCT_NAME}/${SNC_PRODUCT_NAME}.qcow2,size=${CRC_VM_DISK_SIZE} \
+    --boot uefi \
+    --noautoconsole \
+    --wait
```

Comment:
Can we share the virt-install cmdline with `microshift.sh`? They are very similar, but at the same time with unneeded (I think) differences.

Reply:
There is some difference but can try to share but not part of this PR. (some refactor PR).

Reply:
We can/should minimize the differences, and use the same order for the arguments, as many identical arguments as possible, ...

Reply:
Added as different commit to consume it.